pkg/controller/postgres/postgres_controller.go

```diff
@@ -130,10 +130,22 @@ func (r *ReconcilePostgres) Reconcile(request reconcile.Request) (reconcile.Resu
 		// create the postgres instance
 		ps, err := p.CreatePostgres(r.ctx, instance)
 		if err != nil {
+			instance.Status.SecretRef = &v1alpha1.SecretRef{}
+			instance.Status.StatusPhase = v1alpha1.PhaseFailed
+			instance.Status.StatusMessage = "failed to create postgres instance"
```

Comment:
If we let the provider functions return a message, or simply expect them to set the message directly on the struct. We could have very specific information being returned in the message, which could be good.

Reply:
Ok makes sense